torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -382,11 +437,23 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
     GRAPH_DEBUG("After CreateAutodiffSubgraphs\n", *copy);
     size_t idx = 0;
     for (Node* dnode : diff_nodes) {
-      GRAPH_DEBUG("Optimizing diff node ", idx);
+      GRAPH_DEBUG("Optimizing diff node ", idx, " in ", *copy);
+      if (!guardDifferentiableGraph(dnode)) {
+        // if we cannot guard (because of inputs without profiling information),
+        // we re-inline the subgraph and remove the differentiable node
```

Comment:
It would be nice if inputs without profiling information we just didn't guard on, instead of disabling differentiable graphs entirely, maybe as a follow up...